modules/storages/lib/api/v3/file_links/storage_url_helper.rb

```diff
@@ -33,4 +33,65 @@ def storage_url_open(file_link, open_location: false)
 
     "#{file_link.storage.host}/f/#{file_link.origin_id}?openfile=#{location_flag}"
   end
+
+  # rubocop:disable Metrics/AbcSize
+  def make_download_url(file_link:, user:)
```

Comment:
I couldn't grasp what this method really does on first read due to its complexity. I'd break it up at least into:

1. get_oauth_token
2. request_download_url

Reply:
Not sure, I can get the issue here. `get_oauth_token` is already a method of the `ConnectionManager` and cannot get extracted any further.

regarding complexity, I'd challenge the fact, that it is too high :D 

this method as exactly three steps, that are all extracted to separate methods, that all return service results (or alike). The three steps are:
- get oauth token (can get wrong -> error handling)
- make direct download request (can get wrong -> error handling)
- parse result into direct download token (can get wrong -> error handling)
- well and a last step pf piecing the information into the return format, which is IMHO not adding complexity, as it doesn't has logic or error cases.

There is this issue about ServiceResult and complexity. This very method is the one, I wrote first as a monad, reducing the lines by half (as the error handling is extracted). This should also be the showcase for an example tech talk, Andreas and me want to have this week.

Reply:
After discussing after daily, we agreed on simply extracting the three exact steps into separate methods, without introducing new functionality on the `ServiceResult` and without writing a new Service for now.

-> Done